modules/signals/spec/types/with-linked-state.types.spec.ts

```diff
@@ -0,0 +1,158 @@
+      )
+      `;
+
+    expectSnippet(snippet).toFail(
+      /Type 'string' is not assignable to type 'WritableSignal<unknown> | (() => unknown)'./
+    );
+  });
+
+  it('resolves to a normal state signal with automatic linkedSignal', () => {
+    const snippet = `
+      const UserStore = signalStore(
+        { providedIn: 'root' },
+        withLinkedState(() => ({ firstname: () => 'John', lastname: () => 'Doe' }))
+      );
+
+      const userStore = new UserStore();
+
+      const firstname = userStore.firstname;
+      const lastname = userStore.lastname;
+    `;
+
+    expectSnippet(snippet).toSucceed();
+
+    expectSnippet(snippet).toInfer('firstname', 'Signal<string>');
+    expectSnippet(snippet).toInfer('lastname', 'Signal<string>');
+  });
+
+  it('resolves to a normal state signal with manual linkedSignal', () => {
+    const snippet = `
+      const UserStore = signalStore(
+        { providedIn: 'root' },
+        withLinkedState(() => ({
+          firstname: linkedSignal(() => 'John'),
+          lastname: linkedSignal(() => 'Doe')
+        }))
+      );
+
+      const userStore = new UserStore();
+
+      const firstname = userStore.firstname;
+      const lastname = userStore.lastname;
+    `;
+
+    expectSnippet(snippet).toSucceed();
+
+    expectSnippet(snippet).toInfer('firstname', 'Signal<string>');
+    expectSnippet(snippet).toInfer('lastname', 'Signal<string>');
+  });
+
+  it('should set stateSignals as DeepSignal for automatic linkedSignal', () => {
+    const snippet = `
+      const UserStore = signalStore(
+        { providedIn: 'root' },
+        withLinkedState(() => ({
+          user: () => ({ id: 1, name: 'John Doe' }),
+          location: () => ({ city: 'Berlin', country: 'Germany' }),
+        }))
+      );
+
+      const userStore = new UserStore();
+
+      const location = userStore.location;
+      const user = userStore.user;
+    `;
+
+    expectSnippet(snippet).toSucceed();
+
+    expectSnippet(snippet).toInfer(
+      'location',
+      'DeepSignal<{ city: string; country: string; }>'
+    );
+    expectSnippet(snippet).toInfer(
+      'user',
+      'DeepSignal<{ id: number; name: string; }>'
+    );
+  });
+
+  it('should set stateSignals as DeepSignal for manual linkedSignal', () => {
+    const snippet = `
+      const UserStore = signalStore(
+        { providedIn: 'root' },
+        withLinkedState(() => ({
+          user: linkedSignal(() => ({ id: 1, name: 'John Doe' })),
+          location: linkedSignal(() => ({ city: 'Berlin', country: 'Germany' })),
+        }))
+      );
+
+      const userStore = new UserStore();
+
+      const location = userStore.location;
+      const user = userStore.user;
+    `;
+
+    expectSnippet(snippet).toSucceed();
+
+    expectSnippet(snippet).toInfer(
+      'location',
+      'DeepSignal<{ city: string; country: string; }>'
+    );
+    expectSnippet(snippet).toInfer(
+      'user',
+      'DeepSignal<{ id: number; name: string; }>'
+    );
+  });
```

Comment:
can we add a type test to ensure that type inference works correctly when computation functions and linked signals are used together within the same factory? Something like:

```ts
const Store = signalStore(
  withState({ foo: '' }),
  withLinkedState(({ foo }) => ({
    bar: () => foo(),
    baz: linkedSignal(() => foo()),
  })),
);
```

Reply:
Done